waku/v2/protocol/noise/payload.go

```diff
@@ -5,12 +5,13 @@ import (
 
 	n "github.com/waku-org/go-noise"
 	"github.com/waku-org/go-waku/waku/v2/protocol/pb"
+	"google.golang.org/protobuf/proto"
 )
 
 // DecodePayloadV2 decodes a WakuMessage to a PayloadV2
 // Currently, this is just a wrapper over deserializePayloadV2 and encryption/decryption is done on top (no KeyInfo)
 func DecodePayloadV2(message *pb.WakuMessage) (*n.PayloadV2, error) {
-	if message.Version != 2 {
+	if message.GetVersion() != 2 {
```

Comment:
Better to use constant for this. 
I am wondering since message versions are defined in rfc for values 0,1 and 2 , if we should include them as enums in protobuf itself rather than using int.
cc @jm-clius , @alrevuelta

Reply:
Although a good idea to have the client implementation be specific here, I'm not sure we want to overdefine this in the protobuf. Since this is mostly an application-defined field and only interesting to the application, I don't see much value being gained in changing from `int`.

Reply:
WakuMessage.Version is a field defined in the WakuMessage.proto right. 
Since message versions are specified as 0,1 and 2 in rfc https://rfc.vac.dev/spec/14/#payload-encryption , i was wondering if it makes sense to include it as part of protobuf definition.
Am i missing something, when you say this is implementation specific and the field in application-defined?

Reply:
Ah "implementation specific" in that it is populated by the application and only of interest to the application and not on the Waku level. Waku (at a routing layer) does not decrypt or encrypt the messages based on this field value, so does not need to parse this field. I'm not sure that the version field should even be defined in the Message RFC, but rather in various higher-layer specs. In a sense this is similar to the `meta` and `content_topic` fields (populated by the application for its own use), except that both of these are used on the Waku layer to determine the message hash. Version field is simply a helper field for the application.